micromamba/tests/test_create.py

```diff
@@ -661,6 +666,57 @@ def test_create_envs_dirs(tmp_root_prefix: Path, tmp_path: Path):
     assert (tmp_path / env_name / "conda-meta" / "history").exists()
 
 
+@pytest.mark.parametrize("cli_root_prefix", (False, True))
+@pytest.mark.parametrize("user_envs_dirs", (False, True))
+@pytest.mark.parametrize("check_config", (False, True))
+def test_root_prefix_precedence(
+    monkeypatch, tmp_path, user_envs_dirs, cli_root_prefix, check_config
+):
+    """Test for root prefix precedence"""
+
+    envroot = tmp_path / "envroot"
+    envroot_envs = envroot / "envs"
+    cliroot = tmp_path / "cliroot"
+    cliroot_envs = cliroot / "envs"
+    userenv = tmp_path / "userenv"
+    userenv_envs = userenv / "envs"
```

Comment:
Actually, I think we need to properly clarify where those come from and their precedence, even if it is super verbose.

`CONDA_ENVS_DIR` directly points to the equivalent of an `envs` directory, so no need for a distinguishing for this one (my bad).

Also, the case of `envs_dir` from the configuration also needs to be treated I believe.

```suggestion
    # Environments can be created in several places depending, in this order:
    #   - 1. in the folder of `CONDA_ENVS_DIR` if it is set
    #   - 2. the root prefix given by the user's command (e.g. specified via the `-r` option)
    #   - 3. the usual root prefix (set generally by `MAMBA_ROOT_PREFIX`)

    # Given by `CONDA_ENVS_DIR`
    conda_envs_dir_envs = tmp_path / "conda_envs_dir_envs"

    # Given by `MAMBA_ROOT_PREFIX`
    mamba_root_prefix = tmp_path / "envroot"
    mamba_root_prefix_envs = envroot / "envs"

    # Given via the CLI
    cli_provided_root = tmp_path / "cliroot"
    cli_provided_root_envs = cli_provided_root / "envs"
```

If this suggestion is accepted, the rest of the test needs to be adapted accordingly.

Reply:
I've made these changes, added a test for when `envs_dirs` is set in the .condarc file, and made the test a bit clearer (hopefully).

Reply:
Thank you!